lead/tasks.py

```diff
@@ -117,6 +106,46 @@ def _extract_from_lead_core(lead_id):
     return True
 
 
+@shared_task(bind=True, max_retries=10)
+def send_lead_text_to_deepl(self, lead_id, text):
+    lead = Lead.objects.filter(id=lead_id).first()
+    if not lead:
+        return True
+
+    try:
+        data = {
+            'deeper': 1,
+            'group_id': lead.project.id,
+            'text': text,
+        }
+        response = requests.post(DEEPL_CLASSIFY_URL,
+                                 data=data)
+        response_data = response.json()
+        classified_doc_id = response_data.get('id')
+
+        # Get preview
+        preview = LeadPreview.objects.filter(lead=lead).first()
+
+        if not preview:
+            # NOTE: This will be weird because by this time preview should have
+            # been created
+            logger.warn(
+                "Lead(id:{}) preview hasn't been created until deepl request".
+                format(lead_id)
+            )
+            return True
+
+        preview.classified_doc_id = classified_doc_id
+        preview.save()
+        return True
+    except Exception as e:
```

Comment:
This should be probably around the request only.

Reply:
considered.